dnf/package.py

```diff
@@ -79,13 +79,8 @@ def _from_system(self):
 
     @property
     def _from_repo(self):
-        pkgrepo = None
         if self._from_system:
-            pkgrepo = self.base.history.repo(self)
-        else:
-            pkgrepo = {}
-        if pkgrepo:
-            return '@' + pkgrepo
+            return self.base.history.repo(str(self))
```

Comment:
If it provides an empty string it should return self.reponame instead.

Reply:
Something like this could help:

            value = self.base.history.repo(str(self))
            if value:
                return self.base.history.repo(str(self))

Reply:
No it doesn't